src/index.ts

```diff
@@ -209,6 +210,18 @@ const setupTsRunner = (
     }
   }
 
+  if (process.versions.pnp) {
+    try {
+      const resolvedPnp = require.resolve('pnpapi')
+      if (
+        !NODE_OPTIONS?.includes(resolvedPnp) &&
```

Comment:
If the pnp file exists in NODE_OPTIONS then this script was likely run by the yarn node cli and not by the eslint vscode extension. Duplicating the `--require pnp` at this point causes the pnp patch to fail to apply.